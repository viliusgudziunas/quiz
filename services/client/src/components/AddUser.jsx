import React, { useState } from "react";
import axios from "axios";

const AddUser = ({ setUserAdded }) => {
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");

  const addUser = e => {
    e.preventDefault();
    const data = { username, email };
    axios
      .post(`${process.env.REACT_APP_USERS_SERVICE_URL}/users`, data)
      .then(res => {
        setUserAdded(true);
        setUsername("");
        setEmail("");
      })
      .catch(err => {
        console.log(err.response.data);
      });
  };

  return (
    <form onSubmit={addUser}>
      <div className="field">
        <input
          type="text"
          className="input is-large"
          name="username"
          placeholder="Enter a username"
          required
          value={username}
          onChange={e => setUsername(e.target.value)}
        />
      </div>
      <div className="field">
        <input
          type="email"
          className="input is-large"
          name="email"
          placeholder="Enter an email address"
          required
          value={email}
          onChange={e => setEmail(e.target.value)}
        />
      </div>
      <input
        type="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
      />
    </form>
  );
};

export default AddUser;
