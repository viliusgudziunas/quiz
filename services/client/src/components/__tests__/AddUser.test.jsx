import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import AddUser from "../AddUser";

test("AddUser renders properly", () => {
  const wrapper = shallow(<AddUser />);
  const element = wrapper.find("form");
  expect(element.find("input").length).toBe(3);

  const username_field = element.find("input").get(0);
  expect(username_field.props.name).toBe("username");
  expect(username_field.props.value).toBe("");

  const email_field = element.find("input").get(1);
  expect(email_field.props.name).toBe("email");
  expect(email_field.props.value).toBe("");

  const submit_button = element.find("input").get(2);
  expect(submit_button.props.type).toBe("submit");
  expect(submit_button.props.value).toBe("Submit");
});

test("AddUser renders a snapshot properly", () => {
  const tree = renderer.create(<AddUser />).toJSON();
  expect(tree).toMatchSnapshot();
});
