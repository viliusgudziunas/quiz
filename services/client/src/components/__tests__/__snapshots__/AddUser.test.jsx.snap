// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="username"
      onChange={[Function]}
      placeholder="Enter a username"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="email"
      onChange={[Function]}
      placeholder="Enter an email address"
      required={true}
      type="email"
      value=""
    />
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    name="submit"
    type="submit"
    value="Submit"
  />
</form>
`;
